e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,162 @@
+	MetadataV2: &v1.Metadata{
+		CvssV2: nil,
+		CvssV3: &v1.CVSSMetadata{
+			Score:               7,
+			Vector:              "CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H",
+			ExploitabilityScore: 1,
+			ImpactScore:         5.9,
+		},
+	},
+	FixedBy:  "",
+	Severity: "Moderate",
+}
+
+func TestGRPCGetRHCOSNodeVulnerabilities(t *testing.T) {
+	conn := connectToScanner(t)
+	client := v1.NewNodeScanServiceClient(conn)
+
+	cases := []struct {
+		request          *v1.GetNodeVulnerabilitiesRequest
+		responseContains *v1.GetNodeVulnerabilitiesResponse
+	}{
+		{
+			request: &v1.GetNodeVulnerabilitiesRequest{
+				OsImage:          "Ubuntu 20.04.1 LTS",
+				KernelVersion:    "0.0.1",
+				KubeletVersion:   "0.0.1",
+				KubeproxyVersion: "0.0.1",
+				Runtime: &v1.GetNodeVulnerabilitiesRequest_ContainerRuntime{
+					Name:    "docker",
+					Version: "0.0.1",
+				},
+				NodeInventory: &v1.Components{
+					Namespace: "Red Hat CoreOS",
+					RhelComponents: []*v1.RHELComponent{
+						{
+							Id:        int64(1),
+							Name:      "libksba",
+							Namespace: "rhel:8",
+							Version:   "1.3.5-7.el8",
+							Arch:      "x86_64",
+							Module:    "",
+							// From: https://www.redhat.com/security/data/metrics/repository-to-cpe.json
+							// "rhel-8-for-x86_64-appstream-rpms": {"cpes": ["cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream"]},
+							// "rhel-8-for-x86_64-baseos-rpms": {"cpes": ["cpe:/o:redhat:enterprise_linux:8::baseos", "cpe:/o:redhat:rhel:8.3::baseos"]}
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+						{
+							Id:        int64(2),
+							Name:      "tar",
+							Namespace: "rhel:8",
+							Version:   "1.27.1.el8",
+							Arch:      "x86_64",
+							Module:    "",
+							Cpes: []string{
+								"cpe:/a:redhat:enterprise_linux:8::appstream", "cpe:/a:redhat:rhel:8.3::appstream",
+								"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+							},
+							AddedBy: "",
+						},
+					},
+					LanguageComponents: nil,
+				},
+			},
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				InventoryFeatures: []*v1.Feature{
+					{
+						Name:            "libksba",
+						Version:         "1.3.5-7.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnLibksba},
+					},
+					{
+						Name:            "tar",
+						Version:         "1.27.1.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnTar},
+					},
+				},
+			},
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case-%d", i), func(t *testing.T) {
+			c := c
+			resp, err := client.GetNodeVulnerabilities(context.Background(), c.request)
+			require.NoError(t, err)
+			for _, expectedFeat := range c.responseContains.InventoryFeatures {
+				found := false
```

Comment:
don't care too much since this is a test, but usually `var found bool` is preferred